src/Workspaces/CoreTest/Options/OptionSerializerTests.cs

```diff
@@ -0,0 +1,145 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using Microsoft.CodeAnalysis.Completion;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Editor.Implementation.Suggestions;
+using Microsoft.CodeAnalysis.Editor.InlineDiagnostics;
+using Microsoft.CodeAnalysis.Editor.Shared.Options;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.ImplementType;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.SolutionCrawler;
+using Roslyn.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.UnitTests.Options;
+
+public class OptionSerializerTests
+{
+    private const string invalidOption = "Hello I can't be a valid option value";
+
+    [Theory, CombinatorialData]
+    public void SerializationAndDeserializationForNullableBoolean([CombinatorialValues(true, false, null)] bool? value)
+    {
+        var options = new IOption2[]
+        {
+            CompletionViewOptions.EnableArgumentCompletionSnippets,
+            FeatureOnOffOptions.OfferRemoveUnusedReferences,
+            FeatureOnOffOptions.ShowInheritanceMargin,
+            SuggestionsOptions.Asynchronous,
+            WorkspaceConfigurationOptionsStorage.EnableOpeningSourceGeneratedFilesInWorkspace,
+            SolutionCrawlerOptionsStorage.EnableDiagnosticsInSourceGeneratedFiles,
+            CompletionOptionsStorage.ShowItemsFromUnimportedNamespaces,
+            CompletionOptionsStorage.ShowNewSnippetExperienceUserOption,
+            CompletionOptionsStorage.TriggerOnDeletion,
+        };
+
+        foreach (var option in options)
+        {
+            var serializer = option.Definition.Serializer;
+            var serializedValue = serializer.Serialize(value);
+            switch (value)
+            {
+                case null:
+                    Assert.Equal("null", serializedValue);
+                    break;
+                case true:
+                    Assert.Equal("True", serializedValue);
+                    break;
+                case false:
+                    Assert.Equal("False", serializedValue);
+                    break;
+                default:
+                    throw ExceptionUtilities.Unreachable();
+            }
+
+            var success = serializer.TryParse(serializedValue, out var parsedResult);
+            Assert.True(success, $"Can't parse option: {option.Name}, value: {serializedValue}");
+            Assert.Equal(value, parsedResult);
+
+            VerifyInvalidParse(option);
+        }
+    }
+
+    [Fact]
+    public void SerializationAndDeserializationForEnum()
```

Comment:
>SerializationAndDeserializationForEnum [](http://example.com/codeflow?start=16&length=38)

Enum.TryParse has an odd behavior for non-flags enums:

```
> enum F { A = 1, B = 2, C = 4 }
> Enum.TryParse<F>("A, B", out var x)
true
> x
C
> 
``` 

Perhaps we should fail the parsing if the string contains `,` to avoid the above.

Reply:
I just play with it with try dotnet and this behavior of `Enum.TryParse` is so strange

Reply:
I will probably add a test case for this

Reply:
Done